build/accelerate_launch.py

```diff
@@ -47,6 +48,40 @@
 ERROR_LOG = "/dev/termination-log"
 
 
+def copytree(source, destination, symlinks=False, ignore=None):
+    if not os.path.exists(destination):
+        os.makedirs(destination)
+        shutil.copystat(source, destination)
+    file_list = os.listdir(source)
+    # If we want to ignore any files
+    if ignore:
+        exclude = ignore(source, file_list)
+        file_list = [a_file for a_file in file_list if a_file not in exclude]
+    # Have a list of directory objects, now iterate over them.
+    for item in file_list:
+        source_file = os.path.join(source, item)
+        destination_file = os.path.join(destination, item)
+        if symlinks and os.path.islink(source_file):
+            if os.path.lexists(destination_file):
+                os.remove(destination_file)
+            os.symlink(os.readlink(source_file), destination_file)
+            # Improvement over shutil.copytree: BAD CHMOD IS NOT FATAL.
+            try:
+                # get the permissions on the source file
+                st = os.lstat(source_file)
+                mode = stat.S_IMODE(st.st_mode)
+                # apply them to the destination file
+                os.lchmod(destination_file, mode)
+            except Exception as e:  # pylint: disable=broad-except
+                pass  # Not fatal.
+        elif os.path.isdir(source_file):
+            # recursive call for subdirectories
+            copytree(source_file, destination_file, symlinks, ignore)
+        else:
+            # straight copy.
+            shutil.copy2(source_file, destination_file)
```

Comment:
In the issue, we had noted that we didn't want to copy the file permissions over. I don't think the file permissions were the cause of the issue but that's why we were suggesting to use `copyfiles` instead which does not copy the file permissions.

> we should move away from copytree to using copyfiles instead that does not copy permissions
> https://github.com/foundation-model-stack/fms-hf-tuning/blob/main/build/accelerate_launch.py#L127
> copyfiles https://docs.python.org/3/library/shutil.html#shutil.copyfile